riscv-pac/src/lib.rs

```diff
@@ -26,11 +55,34 @@ pub unsafe trait InterruptNumber: Copy {
     fn from_number(value: u16) -> Result<Self, u16>;
 }
 
+/// Marker trait for enums of target-specific core interrupt numbers.
+///
+/// Core interrupts are interrupts are retrieved from the `mcause` CSR. Usually, vectored mode is
+/// only available for core interrupts. The `riscv` crate provides a default implementation for
+/// the RISC-V ISA. However, a PAC may override the default implementation if the target has a
+/// different interrupt numbering scheme (e.g., ESP32C3).
+///
+/// # Safety
+///
+/// Each enum variant must represent a valid core interrupt number read from the `mcause` CSR.
+pub unsafe trait CoreInterruptNumber: InterruptNumber {}
```

Comment:
For interrupt codes that **can be read from xcause**

Reply:
Does this need to be an `unsafe` trait? `mcause` is a `WARL` register, so even if a user attempts to write an invalid value, no soundness guarantees will be violated.

Are there other reasons for making these traits `unsafe`?

Reply:
The idea of making these traits `unsafe` is preventing users from implementing them over any structure. Ideally, only PACs via `svd2rust` should implement them. In this way, peripheral drivers etc. can rely over these traits to ensure that their APIs are safe even using raw pointers behind the scenes.